### PR TITLE
[JENKINS-69489] - Hide restart checkbox if the controller doesn't support it

### PR DESCRIPTION
**Pull request by NotMyFault**
See [JENKINS-69489](https://issues.jenkins.io/browse/JENKINS-69489).

I refactored the existing code a bit to use a Jenkins checkbox rather than labels and inputs and to display the checkbox only, if the controller can be restarted that way through the update center. Instead of displaying a label regardless, the `if` block above.

### Testing done

Looks and works like that on macOS:

Screenshot 2023-03-17 at 09 59 26

Testing on a Windows VM doesn't show the checkbox. As noted by James, Windows controllers don't support that:

Screenshot 2023-03-17 at 10 06 23

### Proposed changelog entries

- Hide 'Restart Jenkins' checkbox in the update center, if the controller doesn't support it.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7740"></a>

### CODE REVIEW

**Review by timja — APPROVED** [review]
nice

**Review by janfaracik — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

File: core/src/main/resources/hudson/model/UpdateCenter/body.jelly
Comment (daniel-beck) [inline core/src/main/resources/hudson/model/UpdateCenter/body.jelly:58]: @NotMyFault Why did you (attempt to) make the checkbox here `readonly`? It doesn't work because https://github.com/jenkinsci/jenkins/commit/4cac031d81b4385410d1081548221ffefd1007c9#diff-6fbee721e02514a54ede8e6e5ed37d3af538239842117a33b7ee905b0a518c81R6 is wrong. Asking because I want to fix my bug and want to ensure I'm not introducing a different one.

Comment (NotMyFault) [reply core/src/main/resources/hudson/model/UpdateCenter/body.jelly:58]: Good question, actually. I _think_ the attribute is obsolete and can be removed, given the if-condition represents the default state and the checkbox should work here.

**Pull request merged**
